g0 done

Gaydarenko opened the pull request:
Tests passed successful - classwork

Gaydarenko committed:
g0 done
Tests passed successful

aeksei commented on Tasks/g0_bubble_sort.py:
```diff
@@ -8,4 +8,8 @@ def sort(container: List[int]) -> List[int]:
     :param container: container of elements to be sorted
     :return: container sorted in ascending order
     """
+    for _ in range(len(container)):
+        for i in range(len(container)-1):
+            if container[i] > container[i+1]:
+                container[i], container[i+1] = container[i+1], container[i]
```
добавьте ещё проверку, что если так случилось, что в середине сортировки последовательность отсортирована (т.е. на очередном проходе не было переставновки элементов), то можно досрочно прекращать цикл

Gaydarenko replied:
Добавил проверку оставшейся части. Но это не экономит время, даже наоборот (ведь каждый раз приходится делать проверку if). Для проверки внес изменения в тесты - увеличил список в сотню раз

Gaydarenko replied:
Но при этом сделал небольшую доработку:
С каждым проходом первого цикла уменьшается размер второго цикла на 1 (т.к. в конце уже отсортированные элементы)

aeksei replied:
@Gaydar-Kanub проверка if выполняется за О(1), а вот сортировка пузырьком O(n^2)
на мылах примерах может и не будет видно выигрыша по времени, а вот при большом количестве данных уже будет заметнее

aeksei reviewed (changes requested):
---

Если стоит статус **Changes requested**, то нужно в данную ветку внести обязательные изменения, которые представлены ниже. Если выше представлены комментарии, то они носят рекомендательный характер.
После нужно сделать "push", тогда коммиты с изменениями сразу попадут в "pull request". 
После этого нужно будет запросить "re-request review"

Gaydarenko committed:
1) С каждым проходом первого цикла уменьшаетсяразмер второго цикла на 1 (т.к. в конце уже отсортированные элементы)
2) Добавил проверку оставшейся части. Но это не экономит время, даже наоборот (ведь каждый раз приходится делать проверку if). Для проверки внес изменения в тесты - увеличил список в сотню раз
